spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/SparkSpaceCurveUDF.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.actions;
+
+import org.apache.spark.sql.Column;
+import org.apache.spark.sql.types.DataType;
+
+public interface SparkSpaceCurveUDF {
+
+  Column sortedLexicographically(Column column, DataType type);
+
+  Column interleaveBytes(Column arrayBinary);
```

Comment:
This I think is a little incorrect, the ZOrder algorithm interleaves the Hilbert doesn't quite do that. Maybe this should just be combineColumns or something like that?

Reply:
Thank you, but I don't understand the meaning here. I think the hilbert value is generated, isn't it hilbert curve?

Reply:
For ZOrder bytes are directly interleaved, AAAA and BBBB -> ABABABA

For Hilbert it's give coordinates AAAA and BBBB what is the corresponding hilbert curve location given a hilbert curve of X bit depth. So that isn't an interleaving.

Reply:
thanks.I'll take a closer look later.

Reply:
> For ZOrder bytes are directly interleaved, AAAA and BBBB -> ABABABA
> 
> For Hilbert it's give coordinates AAAA and BBBB what is the corresponding hilbert curve location given a hilbert curve of X bit depth. So that isn't an interleaving.

I take a look, this method name caused a misunderstanding, in fact, it did not interleaving for hilbert.I'll update it;